tests/subsys/llext/simple/src/threads_kernel_objects_ext.c

```diff
@@ -23,9 +25,12 @@ void test_thread(void *arg0, void *arg1, void *arg2)
 void test_entry(void)
 {
 	printk("Give semaphore from main thread\n");
+	zassert_not_null(&my_sem);
```

Comment:
just to make sure - this won't be optimised out by the compiler?

Reply:
No, since it has no indication about the symbol (other than it's an `extern`), so it will dutifully provide a relocation record for that and test it at runtime. 
Granted, it may be a _different_ relocation type than the call below, depending on how it actually translates to machine code... but covering all relocations is well outside these sanity checks.